components/public-api-server/pkg/oidc/router.go

```diff
@@ -92,8 +93,8 @@ func (s *Service) getCallbackHandler() http.HandlerFunc {
 	return func(rw http.ResponseWriter, r *http.Request) {
 		config, state, err := s.GetClientConfigFromCallbackRequest(r)
 		if err != nil {
-			log.Warn("client config not found: " + err.Error())
-			http.Error(rw, "config not found", http.StatusNotFound)
+			log.WithError(err).Warn("Client SSO config not found")
+			http.Error(rw, "We were unable to find the SSO configuration you've requested. Please ensure your SSO configuration is correct, and validated.", http.StatusNotFound)
```

Comment:
Not every user can fix this situation. 

Suggestion: `SSO config not found. Please contact the admin of your organization.`